src/SecTester.Scan/Scan.cs

```diff
@@ -1,58 +1,160 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using SecTester.Core.Extensions;
 using SecTester.Scan.Models;
 
 namespace SecTester.Scan;
 
-public class Scan : IDisposable
+public class Scan : IAsyncDisposable
 {
+  private static readonly TimeSpan DefaultPollingInterval = TimeSpan.FromSeconds(5);
+
   private static readonly IReadOnlyCollection<ScanStatus> ActiveStatuses =
-    new[]
-    {
-      ScanStatus.Pending, ScanStatus.Running, ScanStatus.Queued
-    };
+    new[] { ScanStatus.Pending, ScanStatus.Running, ScanStatus.Queued };
 
   private static readonly IReadOnlyCollection<ScanStatus> DoneStatuses =
-    new[]
-    {
-      ScanStatus.Disrupted, ScanStatus.Done, ScanStatus.Failed, ScanStatus.Stopped
-    };
+    new[] { ScanStatus.Disrupted, ScanStatus.Done, ScanStatus.Failed, ScanStatus.Stopped };
 
   private readonly ScanOptions _options;
-
+  private readonly ILogger _logger;
   private readonly Scans _scans;
-  private readonly ScanState _scanState = new(ScanStatus.Pending);
+  private ScanState _scanState = new(ScanStatus.Pending);
+  private readonly SemaphoreSlim _semaphore = new(1, 1);
+
+  public string Id { get; }
+
+  private bool DoneCore => DoneStatuses.Contains(_scanState.Status);
+  private bool ActiveCore => ActiveStatuses.Contains(_scanState.Status);
 
-  public Scan(Scans scans, ScanOptions options)
+  public bool Active
   {
+    get
+    {
+      using var _ = _semaphore.Lock();
```

Comment:
As discussed yesterday, it is pointless to lock this for reading. You just need to hold a lock on the assignment due to the immutable nature of `ScanState`. You just need to read a copy to ensure atomicity.